pkgs/development/libraries/mesa/default.nix

```diff
@@ -124,12 +126,16 @@ self = stdenv.mkDerivation {
     "-Dmicrosoft-clc=disabled" # Only relevant on Windows (OpenCL 1.2 API on top of D3D12)
   ] ++ optionals stdenv.isLinux [
     "-Dglvnd=true"
+  ] ++ optionals enableOpenCL [
+    "-Dgallium-opencl=icd"
+    "-Dopencl-spirv=false"
+    "-Dclang-libdir=${llvmPackages.clang-unwrapped.lib}/lib"
   ];
 
   buildInputs = with xorg; [
     expat llvmPackages.libllvm libglvnd xorgproto
     libX11 libXext libxcb libXt libXfixes libxshmfence libXrandr
-    libffi libvdpau libelf libXvMC
+    libffi libvdpau libelf libXvMC libclc llvmPackages.clang llvmPackages.clang-unwrapped
```

Comment:
I think we should only add them with `enableOpenCL` enabled for now (to avoid accidentally depending on them after Mesa updates and to "document" this).